### ISSUE TITLE
Make `BeartypeValidatorTester` great again

### ISSUE THREAD

**Issue by minmax** [beartype/beartype#416]
Now `BeartypeValidatorTester` sets too strict restrictions on `object`, which makes it difficult to use functions with a specific type.

```python
BeartypeValidatorTester = Callable[[object], bool]
```

As I would like to write functions:
```python
@beartype
def foo(ascii: Annotated[str, Is[str.isascii]]) -> str:
   return ascii
```

but `pyright` complains
```python
# Is[str.isascii]
"object" is incompatible with "str" reportArgumentType
```

I can handle error with handmade helper
```python
def is_ascii(val: object) -> bool:
    return isinstance(val, str) and val.isascii()
```

and use it as 
```python
Is[is_ascii]
```
but this `isinstance` check is redundant, it is already done by `@beartype`

so, maybe change `BeartypeValidatorTester` to
```python
BeartypeValidatorTester = Callable[[Any], bool]
```
?

**Comment by leycec** [comment]
**Resolved by b41845692b989334441d.** Excellent! Thanks so much for that detailed writeup, @minmax. As always, I agree with everything you say. You're the min-max QA Chad King, after all. Thankfully, resolving this was *super* easy. If only every feature request was like this. :sob:

Would you mind installing this commit and letting me know if that resolves everything on your end? Just:

```bash
pip install git+https://github.com/beartype/beartype.git@b41845692b989334441d68d77516eb4d890c624f
```

Thanks again! Emoji man bows before you. :bow:

**Comment by minmax** [comment]
> that resolves everything on your end

yes, thank you, there is no more error
